---
title: "5top utr data exploration"
author: "Maddie Lombardo"
date: "3/7/2022"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
library(pacman)
pacman::p_load(annotatr,janitor,tidyverse,AnnotationDbi,org.Hs.eg.db,plyranges,TxDb.Hsapiens.UCSC.hg38.knownGene)
library(corrplot)

```

```{r torin data from AL}
torin_translation <- readxl::read_excel("C:/Users/mlomb/Desktop/tracked_files_github/bdnf_4su/data/torin_mass_spec.xlsx") %>% 
    clean_names() %>% 
    separate_rows(gene_symbol) %>% 
  as_data_frame()
```

building annotations
```{r building annotations}
annotations = build_annotations(genome = "hg38", annotations = c("hg38_basicgenes"))
annotations = keepStandardChromosomes(annotations, pruning.mode = "coarse")
```

reading bed files
```{r reading in bed files}
bed_file_1hr <- "C:/Users/mlomb/Desktop/tracked_files_github/bdnf_4su/data/bdnf-4su_splicing/bdnf-4su_splicing/control1-bdnf1_annotated.junctions.bed"
bed_1hr_peaks <- read_regions(bed_file_1hr)
csv_1hr <- read_csv("C:/Users/mlomb/Desktop/tracked_files_github/bdnf_4su/data/bdnf-4su_splicing/bdnf-4su_splicing/control1-bdnf1_annotated_junctions.csv")

bed_file_2hr <- "C:/Users/mlomb/Desktop/tracked_files_github/bdnf_4su/data/bdnf-4su_splicing/bdnf-4su_splicing/control2-bdnf2_annotated.junctions.bed"
bed_2hr_peaks <- read_regions(bed_file_2hr)
csv_2hr <- read_csv("C:/Users/mlomb/Desktop/tracked_files_github/bdnf_4su/data/bdnf-4su_splicing/bdnf-4su_splicing/control2-bdnf2_annotated_junctions.csv")

bed_file_6hr <- "C:/Users/mlomb/Desktop/tracked_files_github/bdnf_4su/data/bdnf-4su_splicing/bdnf-4su_splicing/control6-bdnf6_annotated.junctions.bed"
bed_6hr_peaks <- read_regions(bed_file_6hr)
csv_6hr <- read_csv("C:/Users/mlomb/Desktop/tracked_files_github/bdnf_4su/data/bdnf-4su_splicing/bdnf-4su_splicing/control6-bdnf6_annotated_junctions.csv")

```

filtering the significant data; standard use is probability_changing >0.9; also filtering to get only the events that increase in bdnf22
```{r filtering igv data}
sig_changes_1hr <- csv_1hr %>% 
  filter(probability_changing > 0.9) %>% 
  filter(mean_dpsi_per_lsv_junction > 0.15) %>% 
  dplyr::select(gene_name, gene_id, paste_into_igv_junction, mean_dpsi_per_lsv_junction, control1_mean_psi,bdnf1_mean_psi,paste_into_igv_junction,
           seqnames,start,end) %>% 
  mutate(start = start + 1) %>% 
  arrange(-mean_dpsi_per_lsv_junction)

sig_changes_2hr <- csv_2hr %>% 
  filter(probability_changing > 0.9) %>% 
  filter(mean_dpsi_per_lsv_junction > 0.15) %>% 
  dplyr::select(gene_name, gene_id, paste_into_igv_junction, mean_dpsi_per_lsv_junction, control2_mean_psi,bdnf2_mean_psi,paste_into_igv_junction,
           seqnames,start,end) %>% 
  mutate(start = start + 1) %>% 
  arrange(-mean_dpsi_per_lsv_junction)

sig_changes_6hr <- csv_6hr %>% 
  filter(probability_changing > 0.9) %>% 
  filter(mean_dpsi_per_lsv_junction > 0.15) %>% 
  dplyr::select(gene_name, gene_id, paste_into_igv_junction, mean_dpsi_per_lsv_junction, control6_mean_psi,bdnf6_mean_psi,paste_into_igv_junction,
           seqnames,start,end) %>% 
  mutate(start = start + 1) %>% 
  arrange(-mean_dpsi_per_lsv_junction)

sig_peaks_1hr <- bed_1hr_peaks %>% 
  filter(start %in% sig_changes_1hr$start) %>% 
  plyranges::arrange(-score)

sig_peaks_2hr <- bed_2hr_peaks %>% 
  filter(start %in% sig_changes_2hr$start) %>% 
  plyranges::arrange(-score)

sig_peaks_6hr <- bed_6hr_peaks %>% 
  filter(start %in% sig_changes_6hr$start) %>% 
  plyranges::arrange(-score)
```

annotating the peaks
```{r annotating peaks}

annotated_peak_1hr <- annotate_regions(regions = GRanges(sig_peaks_1hr), annotations = annotations) %>%  
  as_tibble() %>%
  clean_names()

annotated_peak_2hr <- annotate_regions(regions = GRanges(sig_peaks_2hr), annotations = annotations) %>%  
  as_tibble() %>%
  clean_names()

annotated_peak_6hr <- annotate_regions(regions = GRanges(sig_peaks_6hr), annotations = annotations) %>%  
  as_tibble() %>%
  clean_names()
```

pulling out the 5'UTRs
```{r 5'UTRs pulled}
five_UTR_1hr <- annotated_peak_1hr %>% 
  dplyr::select(seqnames,start,end,name,annot_symbol,annot_type,score) %>% 
  unique() %>% 
  filter(!(annot_type %in% c("hg38_genes_1to5kb", "hg38_genes_promoters"))) %>% 
  filter(annot_type == "hg38_genes_5UTRs") %>% 
  pull(annot_symbol)

five_UTR_2hr <- annotated_peak_2hr %>% 
  dplyr::select(seqnames,start,end,name,annot_symbol,annot_type,score) %>% 
  unique() %>% 
  filter(!(annot_type %in% c("hg38_genes_1to5kb", "hg38_genes_promoters"))) %>% 
  filter(annot_type == "hg38_genes_5UTRs") %>% 
  pull(annot_symbol)

five_UTR_6hr <- annotated_peak_6hr %>% 
  dplyr::select(seqnames,start,end,name,annot_symbol,annot_type,score) %>% 
  unique() %>% 
  filter(!(annot_type %in% c("hg38_genes_1to5kb", "hg38_genes_promoters"))) %>% 
  filter(annot_type == "hg38_genes_5UTRs") %>% 
  pull(annot_symbol)
```


What's the overlap with genes that change signficantly in their splicing and those
that are differentially expressed on total and new rna level?
```{r splicing and rna categorizations}
new_ratio_bayesian <- read_csv("C:/Users/mlomb/Desktop/tracked_files_github/bdnf_4su/data/new_ratio_bayesian_p_de.csv")

overlap_1hr_splice <- new_ratio_bayesian %>% 
  filter(!is.na(gene_name)) %>% 
  filter(time == 1) %>% 
  filter(gene_name %in% five_UTR_1hr) %>% 
  select(new_rna_sig,total_rna_sig) %>% 
  table()

overlap_2hr_splice <- new_ratio_bayesian %>% 
  filter(!is.na(gene_name)) %>% 
  filter(time == 2) %>% 
  filter(gene_name %in% five_UTR_2hr) %>% 
  select(new_rna_sig,total_rna_sig) %>% 
  table()

overlap_6hr_splice <- new_ratio_bayesian %>% 
  filter(!is.na(gene_name)) %>% 
  filter(time == 6) %>% 
  filter(gene_name %in% five_UTR_6hr) %>% 
  select(new_rna_sig,total_rna_sig) %>% 
  table()
```

make chisquare correlation plots for the overlap
```{r corrplots}
chi_sq_1hr <- new_ratio_bayesian %>% 
    filter(new_rna_sig != "unclear") %>% 
    mutate(gene = gsub("\\..*", "", gene)) %>%  
    left_join(annotables::grch38 %>% select(ensgene,symbol), by = c("gene" = "ensgene")) %>% 
    select(time,symbol,total_rna_sig,new_rna_sig) %>% 
    filter(!is.na(symbol)) %>% 
    unique() %>% 
    mutate(five_UTR = symbol %in% five_UTR_1hr) %>% 
    filter(time == 1) %>% 
    mutate(total_new = paste(total_rna_sig,new_rna_sig,sep = "_")) %>% 
    select(total_new,five_UTR)  %>% 
    table() %>% 
    chisq.test()

chi_sq_2hr <- new_ratio_bayesian %>% 
    filter(new_rna_sig != "unclear") %>% 
    mutate(gene = gsub("\\..*", "", gene)) %>%  
    left_join(annotables::grch38 %>% select(ensgene,symbol), by = c("gene" = "ensgene")) %>% 
    select(time,symbol,total_rna_sig,new_rna_sig) %>% 
    filter(!is.na(symbol)) %>% 
    unique() %>% 
    mutate(five_UTR = symbol %in% five_UTR_2hr) %>% 
    filter(time == 2) %>% 
    mutate(total_new = paste(total_rna_sig,new_rna_sig,sep = "_")) %>% 
    select(total_new,five_UTR)  %>% 
    table() %>% 
    chisq.test()

chi_sq_6hr <- new_ratio_bayesian %>% 
    filter(new_rna_sig != "unclear") %>% 
    mutate(gene = gsub("\\..*", "", gene)) %>%  
    left_join(annotables::grch38 %>% select(ensgene,symbol), by = c("gene" = "ensgene")) %>% 
    select(time,symbol,total_rna_sig,new_rna_sig) %>% 
    filter(!is.na(symbol)) %>% 
    unique() %>% 
    mutate(five_UTR = symbol %in% five_UTR_6hr) %>% 
    filter(time == 6) %>% 
    mutate(total_new = paste(total_rna_sig,new_rna_sig,sep = "_")) %>% 
    select(total_new,five_UTR)  %>% 
    table() %>% 
    chisq.test()

corrplot(chi_sq_1hr$residuals, is.cor = FALSE)
corrplot(chi_sq_2hr$residuals, is.cor = FALSE)
corrplot(chi_sq_6hr$residuals, is.cor = FALSE)

corrplot(100*chi_sq_1hr$residuals^2/chi_sq_1hr$statistic,is.corr = FALSE)
corrplot(100*chi_sq_2hr$residuals^2/chi_sq_2hr$statistic,is.corr = FALSE)
corrplot(100*chi_sq_6hr$residuals^2/chi_sq_6hr$statistic,is.corr = FALSE)
```

#How many of the genes with changes in the 5'utrs are TOP motif containing? 