---
title: "grandslam data_deseq tables function, pvalues, pcaplot, scree, eigencor, volcano, log2foldchange graphs"
author: "Maddie Lombardo"
date: "2/15/2022"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
library(PCAtools)
library(gprofiler2)
library(pcaExplorer)
library(clusterProfiler)
library(RColorBrewer)
library(ggpubr)
library(GOfuncR)
```

This function will be organizing this data into 2 tables that can be input into a Deseq
I will be creating a table 1 that will act as the counts data frame that includes the gene and the min2 counts
Table 2 will act as the metadata for the deseq

```{r reading in file path}
grandslam_new_RNA_file_path <- file.path(here::here('data/grandslam_new_rna.csv'))

grandslam_new_rna <- read_csv(grandslam_new_RNA_file_path)
```

```{r filtering out any estimate new credible level > 20%}
grandslam_new_rna_filtered <- grandslam_new_rna %>% 
  filter(estimate_new_credible_interval < 0.2)
```

creating table 1
```{r filter conversions at specified timepoint and make table 1 from it}
create_table_1 <- function(filtered_grandslam_rna, 
                          timepoint = "6"){
    
  filtered_df <- filtered_grandslam_rna %>% 
      filter(time == timepoint) %>% 
      select(gene, min2, SampleID) %>% 
      mutate(min2 = round(min2)) %>% 
      pivot_wider(names_from = SampleID, values_from = min2, values_fill = 0) 
    
    return(filtered_df)
  }
                         
```

creating table 2
```{r make table 2 for metadata table}
create_metatable <- function(table_1){
  
  meta_table <- tibble(sample = table_1 %>%  
                         colnames()) %>% 
    separate(sample, into = c("condition", "well", "time"), remove = FALSE) %>%
    column_to_rownames('sample') %>% 
    mutate(factor_name = factor(condition, levels = c("control","bdnf")))
  
  return(meta_table)
}
```

now for the deseq part
```{r deseq}
library("DESeq2")

min2_deseq <- function(my_table, time){
  
  table_1 <- create_table_1(my_table,time)
  
  table_1 <- table_1 %>% column_to_rownames('gene')
  
  meta_table <- create_metatable(table_1)
  
  dds_min2 <- DESeqDataSetFromMatrix(countData = table_1,
                                   colData = meta_table,
                                   design = ~ factor_name)
  dds_min2 <- DESeq(dds_min2)

  


  res_min2 <- results(dds_min2) %>% 
    as.data.frame() %>% 
    rownames_to_column('ensgene') %>% 
    mutate(ensgene = gsub("\\..*", "", ensgene)) %>%
    left_join(annotables::grch38 %>% select(ensgene,symbol))

return(res_min2)
}
```

testing function on time 2
```{r test new function on time 2}
time2 <- min2_deseq(grandslam_new_rna_filtered, time = 2)

time2 %>% 
  ggplot(mapping = aes(x = pvalue)) + 
  geom_histogram()+
  ggtitle("P Value Chart time 2")
```

using function on time 1 (it works!)
```{r time 1 pvalue chart}
time1 <- min2_deseq(grandslam_new_rna_filtered, time =1)

time1 %>% 
  ggplot(mapping = aes(x = pvalue)) + 
  geom_histogram()+
  ggtitle("P value Chart time 1")
```

time for 
```{r time 6 pvalue chart}

time6 <- min2_deseq(grandslam_new_rna_filtered, time =6)

time6 %>% 
  ggplot(mapping = aes(x = pvalue)) + 
  geom_histogram()+
  ggtitle("Pvalue chart Time 6")
```

make 1 table
```{r squish together} 
de_list <- list(time1,time2,time6) #make a list of whatever you called the 3 results
samp_ids <- c(1,2,6)  #one two six
full_de_min2 <- purrr::map2(de_list, samp_ids, ~cbind(.x, time = .y)) %>% 
    data.table::rbindlist(.)
```

```{r read in bayesian shizz}
bayesian_fp <- file.path("data","new_ratio_bayesian_p_de.csv")
new_ratio_p <- read_csv(bayesian_fp)
```


we making a log2foldchange graph comparing new to total
```{r log2fold new vs total}

full_de_min2 <- full_de_min2 %>% 
  left_join(new_ratio_p %>% 
              mutate(ensgene = gsub("\\..*", "", gene)), 
            by = c('ensgene','time'),
            suffix = c("_total_rna","_new_rna")) %>%
  filter(!is.na(padj_total_rna)) %>% 
 mutate(significant_log2fold_total = padj_total_rna < 0.1) 

my_color_palette = c("#4cbab3", "#ee6352", "#59cd90", "#fac05e", "#3fa7d6")
  
full_de_min2 %>% 
  mutate(plot_name = ifelse(log2FoldChange_total_rna > 3 & padj_total_rna < 0.05, gene_name_total_rna, NA)) %>% 
  ggplot(aes(x = log2FoldChange_total_rna,
             y = log2FoldChange_new_rna,
             fill = significant_log2fold_total, 
             color = significant_log2fold_total))+
  geom_hline(yintercept = 0)+
  geom_vline(xintercept = 0)+
  geom_point(pch = 21)+
  scale_colour_manual(values = my_color_palette)+
  geom_jitter()+
  facet_wrap(vars(time))+
  geom_text_repel(aes(label = plot_name), color = "black")+
  ggpubr::theme_classic2()


##4 - Compare results here to the ```{r categorizing new vs total} in "differential expression on coverted" analysis
```
explaining the graph quadrants:
close to center: these genes do not have large/significant changes in expression
1(top left): these genes have a log 2fold change increase in new rna but a decrease in total rna
2(top right): these genes have both a log2fold change increase in new and total rna
3(bottom right): these genes have an increase change in total rna but a decreased change in new fraction rna
4(bottom left): these genes have both a decrease change in total and new rna

making new rna categorization chart
```{r new rna categorization}
full_de_min2 <- full_de_min2 %>% 
  mutate(new_rna_is_sig = case_when(padj_new_rna < 0.1 & log2FoldChange_new_rna >0 ~"upregulated",
                                    padj_new_rna < 0.1 & log2FoldChange_new_rna < 0 ~"downregulated",
                                    TRUE ~"not_significant"))

full_de_min2 <- full_de_min2 %>% 
  mutate(new_rna_is_sig_looser_crit = case_when(log2FoldChange_new_rna > 0.5 ~"upregulated",
                                    log2FoldChange_new_rna < -0.5 ~"downregulated",
                                    TRUE ~"not_significant"))

counts_by_hr <- full_de_min2 %>% 
    group_by(time, new_rna_is_sig, total_rna_sig_total_rna) %>% 
    summarize(count = n())

counts_by_hr %>% 
    ggplot(aes(x = new_rna_is_sig, y = count, fill = total_rna_sig_total_rna))+
    scale_fill_brewer(palette = "Paired")+
    geom_col() +
    facet_wrap(vars(time),scales = 'free_x')+
    coord_flip()+
    ggtitle("Categorizing New vs total RNA -\n Differential Expression on Labeled RNA") +
    labs(y = "Number of Genes per Timepoint", 
         x = "New RNA Categorization",
         fill = "Total RNA Categorization")+
  theme(axis.text.y = element_text(angle = 45), legend.position = "bottom", plot.title = element_text(hjust = 0.5), legend.text = element_text(size = 8), legend.title = element_text(size =10))
```

pca plots

first by time
```{r biplotting counts per time for new rna}
time_1_counts_table <- create_table_1(grandslam_new_rna_filtered, time = '1') %>% 
  column_to_rownames(var = 'gene') 

colnames(time_1_counts_table) <- toupper(colnames(time_1_counts_table))

time_1_meta <- create_metatable(time_1_counts_table)

time_1_counts_table <- time_1_counts_table[match(rownames(time_1_meta), colnames(time_1_counts_table))]

p_1hr <- PCAtools::pca(time_1_counts_table , metadata = time_1_meta, removeVar = 0.1)

biplot(p_1hr,
       x = "PC1",
       y = "PC2",
       colby = 'condition',
       colkey = c("#0096FF","#FF42FF"),
       legendPosition = 'right',
       gridlines.major = FALSE,
       gridlines.minor = FALSE,
       colLegendTitle = '')


time_2_counts_table <- create_table_1(grandslam_new_rna_filtered, time = '2') %>% 
  column_to_rownames(var = 'gene')

colnames(time_2_counts_table) <- toupper(colnames(time_2_counts_table))

time_2_meta <- create_metatable(time_2_counts_table)

time_2_counts_table <- time_2_counts_table[match(rownames(time_2_meta), colnames(time_2_counts_table))]

p_2hr <- PCAtools::pca(time_2_counts_table , metadata = time_2_meta, removeVar = 0.1)

biplot(p_2hr,
       x = "PC1",
       y = "PC2",
       colby = 'condition',
       colkey = c("#0096FF","#FF42FF"),
       legendPosition = 'right',
       gridlines.major = FALSE,
       gridlines.minor = FALSE,
       colLegendTitle = '')

time_6_counts_table <- create_table_1(grandslam_new_rna_filtered, time = '6') %>% 
  column_to_rownames(var = 'gene')

colnames(time_6_counts_table) <- toupper(colnames(time_6_counts_table))

time_6_meta <- create_metatable(time_6_counts_table)

time_6_counts_table <- time_6_counts_table[match(rownames(time_6_meta), colnames(time_6_counts_table))]

p_6hr <- PCAtools::pca(time_6_counts_table, metadata = time_6_meta, removeVar = 0.1)

biplot(p_6hr,
       x = "PC1",
       y = "PC2",
       colby = 'condition',
       colkey = c("#0096FF","#FF42FF"),
       legendPosition = 'right', 
       gridlines.major = FALSE,
       gridlines.minor = FALSE,
       colLegendTitle = '')
```

now altogether
```{r squished new rna pca plot}
time_1_counts_table <- time_1_counts_table %>% 
  rownames_to_column('gene')

time_2_counts_table <- time_2_counts_table %>% 
  rownames_to_column('gene')

time_6_counts_table <- time_6_counts_table %>% 
  rownames_to_column('gene')

all_time_points_counts <- left_join(time_1_counts_table, time_2_counts_table, by = 'gene')

all_time_points_counts <- left_join(all_time_points_counts, time_6_counts_table, by = 'gene')

all_time_points_counts <- all_time_points_counts %>% 
  column_to_rownames(var = 'gene')

all_time_points_meta = create_metatable(all_time_points_counts)

all_time_points_counts <- all_time_points_counts[match(rownames(all_time_points_meta), colnames(all_time_points_counts))]

all_time_points_counts[is.na(all_time_points_counts)] <- 0

p_all_times <- PCAtools::pca(all_time_points_counts, metadata = all_time_points_meta, removeVar = 0.1)

biplot(p_all_times,
       x = "PC1",
       y = "PC2",
       colby = 'condition',
       colkey = c("#0096FF","#FF42FF"),
       shape = 'time',
       legendPosition = 'right',
       gridlines.major = FALSE,
       gridlines.minor = FALSE,
       colLegendTitle = '')
```

screeplots new rna
```{r screeplots for time and altogether (smooshed)}
screeplot(p_1hr, title = '', gridlines.major = FALSE, gridlines.minor = FALSE)

screeplot(p_2hr,title = '', gridlines.major = FALSE, gridlines.minor = FALSE)

screeplot(p_6hr,title = '', gridlines.major = FALSE, gridlines.minor = FALSE)

screeplot(p_all_times,title = '', gridlines.major = FALSE, gridlines.minor = FALSE)
```

eigencor plots for new rna
```{r eigencor plots for altogether (smooshed)}
eigencorplot(p_all_times, metavars = c('time', 'well','condition'), colCorval = 'white')
```

making volcano plots for new rna 1, 2, 6 hrs
```{r making volcano plots for new rna 1,2,6hr}
source(here::here('make_volcano_plot.R'))

label_significant(time1, log2FoldCut = 2, log10padj= 10)+
  ggpubr::theme_classic2()

label_significant(time2, log2FoldCut = 2, log10padj= 10)+
  ggpubr::theme_classic2()

label_significant(time6, log2FoldCut = 2, log10padj= 10)+
  ggpubr::theme_classic2()
```

pulling out genes from 2hr that have log2foldchange_total < 0 and log2foldchange_new >0 and making venn diagram

```{r venn diagram of genes that have log2fold_total <0 and log2fold_new >0}
library(eulerr)

venn_genes_quad1 <- full_de_min2 %>% 
    filter(log2FoldChange_total_rna < 0 & log2FoldChange_new_rna > 0) %>%
    filter(time == 2) %>% 
    pull(gene_name) 

venn_genes_bdnf_higher_new_total_downreg <- full_de_min2 %>% 
    filter(new_rna_sig == 'bdnf_higher_new_rna' & total_rna_sig == 'downregulated') %>% 
    filter(time == 2) %>% 
    pull(gene_name) 

venn_table = tibble(gene_names = union(venn_genes_quad1,venn_genes_bdnf_higher_new_total_downreg))

venn_table <- venn_table %>% 
    mutate(quad1_new_rna = gene_names %in% venn_genes_quad1) %>% 
    mutate(bdnf_high_new_total_downreg = gene_names %in% venn_genes_bdnf_higher_new_total_downreg) %>% 
  column_to_rownames('gene_names')*1

fit <- euler(venn_table)

plot(fit)
```

going to look at controls 1 vs 6hr and bdnf 1 vs 6hr
```{r de controls 1vs6hr and bdnf 1vs6hr}
source(here::here('run_standard_deseq.R'))

feature_counts_fp <- file.path(here::here('data','feature_counts_bdnf_No4'))

one_vs_6_dds_control <- run_standard_deseq(feature_counts_fp, 
base_grep = "1hr",
                              contrast_grep = "6hr", 
                              grep_pattern = "CONTROL_.*.*_1|CONTROL_.*.*_6",
                              baseName = "onehour",
                              contrastName = '6hr')

one_vs_6_dds_bdnf <- run_standard_deseq(feature_counts_fp, 
base_grep = "1hr",
                              contrast_grep = "6hr", 
                              grep_pattern = "BDNF_.*.*_1|BDNF_.*.*_6",
                              baseName = "onehour",
                              contrastName = '6hr')
                          
#volcano plots
source(here::here('make_volcano_plot.R'))

label_significant(one_vs_6_dds_control$results_table, log2FoldCut = 1.5, log10padj = 4)+
  scale_fill_manual(values = c("#3fa7d6", "#ee6352", "#fac05e"))+
  scale_color_manual(values = c("#3fa7d6", "#ee6352", "#fac05e"))+
  ggtitle("Volcano 1 vs 6hr Control")+
  ggpubr::theme_pubr()

label_significant(one_vs_6_dds_bdnf$results_table, log2FoldCut =2, log10padj = 6)+
  scale_fill_manual(values = c("#3fa7d6", "#ee6352", "#fac05e"))+
  scale_color_manual(values = c("#3fa7d6", "#ee6352", "#fac05e"))+
  ggtitle("Volcano 1 vs 6hr BDNF")+
  ggpubr::theme_pubr()
```

1vs6 hr venn diagrams for control vs bdnf upreg/downreg
```{r 1vs6 venn diagram upreg/downreg}
library("ggVennDiagram")

upreg_genes_6hr_bdnf <- one_vs_6_dds_bdnf$results_table %>% 
  select(log2FoldChange,padj,gene_name) %>% 
  filter(log2FoldChange > 0.1 & padj < 0.1)

upreg_genes_6hr_control <- one_vs_6_dds_control$results_table %>% 
  select(log2FoldChange,padj,gene_name) %>% 
  filter(log2FoldChange > 0.1 & padj < 0.1)

upreg_gene_6hr_list <- list(A = upreg_genes_6hr_bdnf$gene_name,
                            B = upreg_genes_2hr_control$gene_name)

ggVennDiagram(upreg_gene_6hr_list, category.names = c("BDNF", "Control"))+
  ggtitle("Upregulated Genes 6hr BDNF vs Control")

downreg_genes_6hr_bdnf <-one_vs_6_dds_bdnf$results_table %>% 
  select(log2FoldChange,padj,gene_name) %>% 
  filter(log2FoldChange < -0.1 & padj < 0.1)

downreg_genes_6hr_control <- one_vs_6_dds_control$results_table %>% 
  select(log2FoldChange,padj,gene_name) %>% 
  filter(log2FoldChange < -0.1 & padj < 0.1)

downreg_gene_6hr_list <- list(A = downreg_genes_6hr_bdnf$gene_name,
                            B = downreg_genes_6hr_control$gene_name)

ggVennDiagram(downreg_gene_6hr_list, category.names = c("BDNF", "Control"))+
  ggtitle("Downregulated Genes 6hr BDNF vs Control")
```


now going to 1vs 2 hour for controls and bdnf
```{r 1vs2 hr controls and bdnf}
source(here::here('run_standard_deseq.R'))

feature_counts_fp <- file.path(here::here('data','feature_counts_bdnf_No4'))

one_vs_2_dds_control <- run_standard_deseq(feature_counts_fp, 
base_grep = "1hr",
                              contrast_grep = "2hr", 
                              grep_pattern = "CONTROL_.*.*_1|CONTROL_.*.*_2",
                              baseName = "onehour",
                              contrastName = '2hr')

one_vs_2_dds_bdnf <- run_standard_deseq(feature_counts_fp, 
base_grep = "1hr",
                              contrast_grep = "2hr", 
                              grep_pattern = "BDNF_.*.*_1|BDNF_.*.*_2",
                              baseName = "onehour",
                              contrastName = '2hr')
                          
#volcano plots
source(here::here('make_volcano_plot.R'))

label_significant(one_vs_2_dds_control$results_table, log2FoldCut = 1.5, log10padj = 4)+
  scale_fill_manual(values = c("#3fa7d6", "#ee6352", "#fac05e"))+
  scale_color_manual(values = c("#3fa7d6", "#ee6352", "#fac05e"))+
  ggtitle("Volcano 1 vs 2hr Control")+
  ggpubr::theme_pubr()

label_significant(one_vs_2_dds_bdnf$results_table, log2FoldCut =2, log10padj = 6)+
  scale_fill_manual(values = c("#3fa7d6", "#ee6352", "#fac05e"))+
  scale_color_manual(values = c("#3fa7d6", "#ee6352", "#fac05e"))+
  ggtitle("Volcano 1 vs 2hr BDNF")+
  ggpubr::theme_pubr()
```

gonna make venn diagrams of BDNF 2 hours v Control 2 hours (total RNA) up regulated, downregulated

```{r venn diagrams 2hr bdnf vs 2hr control}
library("ggVennDiagram")

upreg_genes_2hr_bdnf <- one_vs_2_dds_bdnf$results_table %>% 
  select(log2FoldChange,padj,gene_name) %>% 
  filter(log2FoldChange > 0 & padj < 0.1)

upreg_genes_2hr_control <- one_vs_2_dds_control$results_table %>% 
  select(log2FoldChange,padj,gene_name) %>% 
  filter(log2FoldChange > 0.1 & padj < 0.1)

upreg_gene_2hr_list <- list(A = upreg_genes_2hr_bdnf$gene_name,
                            B = upreg_genes_2hr_control$gene_name)

ggVennDiagram(upreg_gene_2hr_list, category.names = c("BDNF", "Control"))+
  ggtitle("Upregulated Genes 2hr BDNF vs Control")

downreg_genes_2hr_bdnf <-one_vs_2_dds_bdnf$results_table %>% 
  select(log2FoldChange,padj,gene_name) %>% 
  filter(log2FoldChange < -0.1 & padj < 0.1)

downreg_genes_2hr_control <- one_vs_2_dds_control$results_table %>% 
  select(log2FoldChange,padj,gene_name) %>% 
  filter(log2FoldChange < -0.1 & padj < 0.1)

downreg_gene_2hr_list <- list(A = downreg_genes_2hr_bdnf$gene_name,
                            B = downreg_genes_2hr_control$gene_name)

ggVennDiagram(downreg_gene_2hr_list, category.names = c("BDNF", "Control"))+
  ggtitle("Downregulated Genes 2hr BDNF vs Control")
```

looking at effect of 4su on rna expression using GO analysis

1vs6 control upregulated
```{r go on 1vs6 hr control: upregulated}
bg_ids <- one_vs_6_dds_control$results_table$Geneid %>% 
  gsub("\\..*", "", .) %>% 
  unique() %>% 
  clipr::write_clip()

upreg_1vs6_control <- one_vs_6_dds_control$results_table %>% 
  filter(log2FoldChange > 0.5 & padj < 0.1) %>% 
  pull(Geneid) %>% 
  gsub("\\..*", "", .) %>% 
  clipr::write_clip()

#graphing the string data
bp_go_1vs6_upreg_fp <- file.path("data", "bp go 1vs6 control upreg.tsv")
bp_go_1vs6_upreg <- janitor::clean_names(read_tsv(bp_go_1vs6_upreg_fp))

bp_go_1vs6_upreg %>% slice_max(strength, n = 10) %>% 
  mutate(term_description = fct_reorder(term_description,strength)) %>% 
  ggplot(aes(y = strength, fill = false_discovery_rate, x = term_description,size = observed_gene_count))+
  geom_point(pch = 21) +
  scale_fill_gradient(low = 'blue', high = 'red')+
  coord_flip()+ 
  theme_pubr()+
  theme(legend.position = 'right')+
  ggtitle("biological process GO on 1vs6_upreg genes")

cc_go_1vs6_upreg_fp <- file.path("data", "cc go 1vs6 control upreg.tsv")
cc_go_1vs6_upreg <- janitor::clean_names(read_tsv(cc_go_1vs6_upreg_fp))

cc_go_1vs6_upreg %>% slice_max(strength, n = 10) %>% mutate(term_description = fct_reorder(term_description,strength)) %>%  ggplot(aes(y = strength, fill = false_discovery_rate, x = term_description,size = observed_gene_count)) + geom_point(pch = 21) + scale_fill_gradient(low = 'blue', high = 'red')+coord_flip()+ theme_pubr()+theme(legend.position = 'right')+
  ggtitle("Cellular Compartment GO on 1vs6_upreg genes")

mf_go_1vs6_upreg_fp <- file.path("data", "mf go 1vs6 control upreg.tsv")
mf_go_1vs6_upreg <- janitor::clean_names(read_tsv(mf_go_1vs6_upreg_fp))

mf_go_1vs6_upreg %>% slice_max(strength, n = 10) %>% mutate(term_description = fct_reorder(term_description,strength)) %>%  ggplot(aes(y = strength, fill = false_discovery_rate, x = term_description,size = observed_gene_count)) + geom_point(pch = 21) +scale_fill_gradient(low = 'blue', high = 'red')+ coord_flip()+ theme_pubr()+ theme(legend.position = 'right')+ggtitle("Molecular function GO on 1vs6_upreg genes")
```

1vs6 control downregulated
```{r go on 1vs6 hr control: downregulated}
downreg_1vs6_control <- one_vs_6_dds_control$results_table %>% 
  filter(log2FoldChange < -0.5 & padj < 0.1) %>% 
  pull(Geneid) %>% 
  gsub("\\..*", "", .) %>% 
  clipr::write_clip()

cc_go_1vs6_downreg_fp <- file.path("data", "cc go 1vs6 control downreg.tsv")
cc_go_1vs6_downreg <- janitor::clean_names(read_tsv(cc_go_1vs6_downreg_fp))

cc_go_1vs6_downreg %>% slice_max(strength, n = 10) %>% mutate(term_description = fct_reorder(term_description,strength)) %>%  ggplot(aes(y = strength, fill = false_discovery_rate, x = term_description,size = observed_gene_count)) + geom_point(pch = 21) + scale_fill_gradient(low = 'blue', high = 'red')+ coord_flip()+ theme_pubr()+ theme(legend.position = 'right')+
  ggtitle("Cellular Compartment GO on 1vs6 downreg genes")
```

going to look at density plots of the new rna at 1,2,6hrs for controls and bdnf
```{r density plots grandslam data}
grandslam_new_rna_1hr <- grandslam_new_rna %>% 
  filter(time == 1)

ggplot(grandslam_new_rna_1hr, aes(fraction_new_rna,fill = condition))+
  geom_density(alpha = 0.4)+
  scale_fill_manual(values = c("#0096FF","#FF42FF"))+
  ggtitle("Density plot of New RNA at 1hr")

grandslam_new_rna_2hr <- grandslam_new_rna %>% 
  filter(time == 2)

ggplot(grandslam_new_rna_2hr, aes(fraction_new_rna, fill = condition))+
  geom_density(alpha = 0.4)+
  scale_fill_manual(values = c("#0096FF","#FF42FF"))+
  ggtitle("Density plot of new RNA at 2hr")

grandslam_new_rna_6hr <- grandslam_new_rna %>% 
  filter(time == 6)


ggplot(grandslam_new_rna_6hr, aes(fraction_new_rna, fill = condition))+
  geom_density(alpha = 0.4)+  
  scale_fill_manual(values = c("#0096FF","#FF42FF"))+
  ggtitle("Density plot of new RNA at 6hr")

```


```{r looking at dynein related genes}
dyn_cc_1vs6_upreg <- cc_go_1vs6_upreg %>% filter(grepl('dyn', term_description,ignore.case = TRUE)) %>% pull(matching_proteins_in_your_network_labels) %>% 
  str_split(.,',')

dyn_mf_1vs6_upreg <- mf_go_1vs6_upreg %>% filter(grepl('dyn',term_description,ignore.case = TRUE)) %>% 
  pull(matching_proteins_in_your_network_labels) %>% 
  str_split(.,',') 

dyn_1vs6_control <- unique(dyn_cc_1vs6_upreg, dyn_mf_1vs6_upreg[[1]] )[[1]]

normalized_counts <- one_vs_6_dds_control$deseq_obj %>% 
  counts(., normalized=TRUE) %>% 
  as.data.frame() %>% 
  rownames_to_column('ensgene') %>% 
  mutate(ensgene = gsub("\\..*", "",ensgene)) %>% 
  left_join(annotables::grch38 %>% select(ensgene,symbol)) %>% 
  filter(symbol %in% dyn_1vs6_control)

normalized_counts <- normalized_counts %>% 
  melt(id.vars = c('ensgene','symbol'))

normalized_counts <- normalized_counts %>% 
  mutate(condition = case_when(grepl('_1hr', variable, ignore.case = TRUE) ~ '1',
    grepl('_2hr', variable, ignore.case = TRUE) ~ '2',
    grepl('_6hr', variable, ignore.case = TRUE) ~ '6'))
   
ggplot(normalized_counts, mapping = aes(condition, value))+
  geom_boxplot()+
  geom_point()+
  geom_text_repel(mapping = aes(label = variable), size = 2.5)+
  facet_wrap(vars(symbol), scales = 'free_y')+
  ggpubr::theme_pubr()+
  labs(x = "Timepoint 1hr vs 6hr", y ="Normalized Counts" )
```

```{r scattterplot log2foldchange total rna}
up_4su_control_6hr <- one_vs_6_dds_control$results_table %>% filter(padj < 0.1 & log2FoldChange > 0)

test = list(up_4su_control_6hr = up_4su_control_6hr$gene_name,upreg_genes_2hr_bdnf = upreg_genes_2hr_bdnf$gene_name)
ggVennDiagram(test, category.names = c("Upregulated by 4SU (6 v 1hr Controls)","Upregulated by BDNF 2hr")) + ggtitle("Effect of 4SU treatment on total gene expression") +
    scale_color_manual(values = c("black","black"))

new_ratio_bayesian_fp <- file.path(here::here('data', 'new_ratio_bayesian_p_de.csv'))
new_ratio_bayesian <- read_csv(new_ratio_bayesian_fp)

new_rna_sig_bit <- new_ratio_bayesian %>% 
  select(gene, new_rna_sig)

names(new_rna_sig_bit)[names(new_rna_sig_bit) == 'gene'] <- 'Geneid'

one_vs_6_dds_control$results_table %>% 
    full_join(one_vs_2_dds_bdnf$results_table, by = c("Geneid"),
              suffix = c("_4su_effect","_BDNF_2hr")) %>% 
  left_join(new_rna_sig_bit, by = c('Geneid')) %>% 
    ggplot(aes(x = log2FoldChange_BDNF_2hr,
               y = log2FoldChange_4su_effect, fill = new_rna_sig, color = new_rna_sig)) + 
    geom_hline(yintercept = 0) +
    geom_vline(xintercept = 0) +
    geom_point(alpha = 0.4) + 
  facet_wrap(vars(new_rna_sig))+
    ggpubr::theme_pubr() +
    ggtitle("Effect on Total RNA with 4SU Treatment(ctrl 6 v 1 hr) vs BDNF 2hr",subtitle = 'all genes') + 
    ylab("Log2Fold Total RNA - 4SU Treatment") +
    xlab("Log2Fold Total RNA- BDNF 2hr")
```

going to add column: bdnf+4su log2foldchange - 4su log2fold change to see if bdnf addition amplifies some of the seen 4su response
```{r bdnf+4su log2foldchange - 4su log2fold change}

one_vs_2_dds_bdnf_results <- one_vs_2_dds_bdnf$results_table %>% 
  full_join(one_vs_6_dds_control$results_table, by = c("Geneid"),
              suffix = c("_BDNF_2hr","_4su_effect_control"))

one_vs_2_dds_bdnf_results <- one_vs_2_dds_bdnf_results %>% 
  mutate(log2fold_ratio = log2FoldChange_4su_effect_control/log2FoldChange_BDNF_2hr) %>% 
  filter(!gene_name_BDNF_2hr == 'AC126474.1') %>% 
  filter(!(log2FoldChange_BDNF_2hr < 0 & log2FoldChange_4su_effect_control > 0)) %>% 
  filter(!(log2FoldChange_BDNF_2hr > 0 & log2FoldChange_4su_effect_control < 0)) %>% 
  mutate(log2fold_ratio = ifelse(log2FoldChange_BDNF_2hr<0, log2fold_ratio*-1, log2fold_ratio))

ggplot(one_vs_2_dds_bdnf_results, aes(log2fold_ratio))+
         geom_density()+
  xlim(-25,25)+
  ggpubr::theme_pubr()+
  ggtitle("Relative gene expression effect of 4su vs BDNF in total RNA")+
  xlab("log2FoldChange_4su / log2FoldChange_BDNF")

background_genes_overlap <- unique(c(one_vs_6_dds_control$results_table$Geneid,one_vs_2_dds_bdnf$results_table$Geneid)) %>% 
    gsub("\\..*", "", .) %>% 
  clipr::write_clip()

one_vs_2_dds_bdnf_results %>% 
  arrange(-log2fold_difference) %>% 
  slice_max(order_by = log2fold_difference, n = 2000) %>% 
  pull(Geneid) %>% 
  gsub("\\..*", "", .) %>% 
  clipr::write_clip()

BP_GO_2hr_bdnf_minus_4su_fp <- file.path("data", "2bdnf-4su_GO_BP.tsv")
BP_GO_2hr_bdnf_minus_4su <- janitor::clean_names(read_tsv(BP_GO_2hr_bdnf_minus_4su_fp))

BP_GO_2hr_bdnf_minus_4su %>% slice_max(strength, n = 10) %>% 
  mutate(term_description = fct_reorder(term_description,strength)) %>% 
  ggplot(aes(y = strength, fill = false_discovery_rate, x = term_description,size = observed_gene_count))+
  geom_point(pch = 21) +
  scale_fill_gradient(low = 'blue', high = 'red')+
  coord_flip()+ 
  theme_pubr()+
  theme(legend.position = 'right')+
  ggtitle("The effect of BDNF accounting for 4su changes", subtitle = "BP GO terms for BDNF +4su at 2hr minus 4su at 6hr")

CC_GO_2hr_bdnf_minus_4su_fp <- file.path("data", "2bdnf-4su_GO_CC.tsv")
CC_GO_2hr_bdnf_minus_4su <- janitor::clean_names(read_tsv(CC_GO_2hr_bdnf_minus_4su_fp))

CC_GO_2hr_bdnf_minus_4su %>% slice_max(strength, n = 10) %>% mutate(term_description = fct_reorder(term_description,strength)) %>%  ggplot(aes(y = strength, fill = false_discovery_rate, x = term_description,size = observed_gene_count)) + geom_point(pch = 21) + scale_fill_gradient(low = 'blue', high = 'red')+coord_flip()+ theme_pubr()+theme(legend.position = 'right')+
  ggtitle("The effect of BDNF accounting for 4su changes", subtitle = "BP GO terms for BDNF +4su at 2hr minus 4su at 6hr")

mf_go_1vs6_upreg_fp <- file.path("data", "mf go 1vs6 control upreg.tsv")
mf_go_1vs6_upreg <- janitor::clean_names(read_tsv(mf_go_1vs6_upreg_fp))

mf_go_1vs6_upreg %>% slice_max(strength, n = 10) %>% mutate(term_description = fct_reorder(term_description,strength)) %>%  ggplot(aes(y = strength, fill = false_discovery_rate, x = term_description,size = observed_gene_count)) + geom_point(pch = 21) +scale_fill_gradient(low = 'blue', high = 'red')+ coord_flip()+ theme_pubr()+ theme(legend.position = 'right')+ggtitle("Molecular function GO on 1vs6_upreg genes")

```

