---
title: "isoform switcher"
author: "Maddie Lombardo"
date: "5/10/2022"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
library(IsoformSwitchAnalyzeR)
```

I'm going to use the vignette work flow for this, step by step, explaining what each bit does

1hr
------------------------------------------------

setup: getting the data you need to do the rest
```{r setup, 1hr}
salmon_data_1hr <- importIsoformExpression(parentDir = "C:/Users/mlomb/Desktop/tracked_files_github/bdnf_4su/data/Full BDNF 4SU - salmon/Full BDNF 4SU - salmon/one_hour",addIsofomIdAsColumn = TRUE)

conditions <- gsub('_[1-9][A-Z]_.*', '', colnames(salmon_data_1hr$abundance)[-1])
conditions <- gsub('[1-9][A-Z]_','', conditions)

myDesign_1hr <- data.frame(
  sampleID = colnames(salmon_data_1hr$abundance)[-1],
  condition = conditions)

comparison_to_make = data.frame(condition_1 = "CONTROL", condition_2 = "BDNF")

aSwitchList_1hr <- importRdata(
  isoformCountMatrix = salmon_data_1hr$counts,
  isoformRepExpression = salmon_data_1hr$abundance,
  designMatrix = myDesign_1hr,
  comparisonsToMake = comparison_to_make,
  isoformExonAnnoation = "C:/Users/mlomb/Desktop/tracked_files_github/bdnf_4su/gencode.v40.annotation.gtf.gz",
  isoformNtFasta = "C:/Users/mlomb/Desktop/tracked_files_github/bdnf_4su/gencode.v40.transcripts.fa",
  showProgress = FALSE
)
```


step 1) extract isoform switches and their sequences
```{r step 1: extract isoform switches 1hr}
aSwitchList_1hr <- readRDS('aSwitchList_1hr.rds')

switch_list_1hr <- isoformSwitchAnalysisPart1(
  switchAnalyzeRlist = aSwitchList_1hr,
  pathToOutput = "C:/Users/mlomb/Desktop/tracked_files_github/bdnf_4su/data/Full BDNF 4SU - salmon/Full BDNF 4SU - salmon",
  outputSequences = TRUE,
  prepareForWebServers = FALSE
)

```

pulling the top 10 isoform changes and making some volcano plots
```{r top isoform switches and volcano plots 1hr}

source(here::here('make_volcano_plot.R'))
library(ggplot2)

switch_list_1hr_filtered <- preFilter(switch_list_1hr,
                                      isoformExpressionCutoff = 1.2) # preFilter to remove lowly expressed features

switch_list_1hr_analyzed <- isoformSwitchTestDEXSeq(
    switchAnalyzeRlist = switch_list_1hr_filtered,
    reduceToSwitchingGenes=TRUE)

switch_list_info_1hr_analyze <- switch_list_1hr_analyzed$isoformSwitchAnalysis

switch_list_info_1hr_analyze <- switch_list_info_1hr_analyze %>% 
  mutate(isoform_id = gsub('\\..*',"", isoform_id)) %>% 
  left_join(annotables::grch38_tx2gene, by = c(isoform_id = 'enstxp')) %>% 
  left_join(annotables::grch38 %>%  dplyr::select(symbol, ensgene), by = 'ensgene')

label_significant(switch_list_info_1hr_analyze, log2FoldCut = 0.2)
```


step 2) plot all isoform switches and their annotation
```{r step 2: plot isoform switches, 1hr}


switch_list_1hr_filtered <- preFilter(switch_list_1hr,
                                      isoformExpressionCutoff = 1.2) # preFilter to remove lowly expressed features

switch_list_1hr_analyzed <- isoformSwitchTestDEXSeq(
    switchAnalyzeRlist = switch_list_1hr_filtered,
    reduceToSwitchingGenes=TRUE)

switch_list_1hr_pfam <- extractSequence(switch_list_1hr_analyzed,
                                        pathToOutput = "C:/Users/mlomb/Desktop/tracked_files_github/bdnf_4su/data/Full BDNF 4SU - salmon/Full BDNF 4SU - salmon",
                                        writeToFile = TRUE)


#isoform_switch_1hr_step2 <- isoformSwitchAnalysisPart2(switchAnalyzeRlist = switch_list_1hr_filtered,
#                                                       n = 10,
#                                                       removeNoncodinORFs = TRUE,
#                                                       pathToCPC2resultFile = "C:/Users/mlomb/Desktop/tracked_files_github/bdnf_4su/data/Full BDNF 4SU - salmon/Full BDNF 4SU - salmon/cpc2_isoform_switch.txt",
#                                                      pathToPFAMresultFile = "C:/Users/mlomb/Desktop/tracked_files_github/bdnf_4su/data/Full BDNF 4SU - salmon/Full #BDNF 4SU - salmon/pfam_analysis.txt",
#                                                       pathToSignalPresultFile = "C:/Users/mlomb/Desktop/tracked_files_github/bdnf_4su/data/Full BDNF 4SU - salmon/Full #BDNF 4SU - salmon/signalIP_1hr.txt",
#                                                       pathToIUPred2AresultFile = "C:/Users/mlomb/Desktop/tracked_files_github/bdnf_4su/data/Full BDNF 4SU - #salmon/Full BDNF 4SU - salmon/IUPred_1hr.result",
#                                                       outputPlots = TRUE,
#                                                       pathToOutput = "C:/Users/mlomb/Desktop/tracked_files_github/bdnf_4su/data/Full BDNF 4SU - salmon/Full BDNF 4SU - salmon/one_hour",
#                                                       fileType = 'pdf')
```

```{r pulling apart part 2 analysis, 1hr}
switch_list_1hr_part2 <- analyzeCPC2(switchAnalyzeRlist = switch_list_1hr,
                                  pathToCPC2resultFile = "C:/Users/mlomb/Desktop/tracked_files_github/bdnf_4su/data/Full BDNF 4SU - salmon/Full BDNF 4SU - salmon/cpc2_1hr.txt",
                                  codingCutoff = 0.5,
                                  removeNoncodinORFs = TRUE)
switch_list_1hr_part2 <- analyzePFAM(switchAnalyzeRlist = switch_list_1hr_part2,
                                  pathToPFAMresultFile = "C:/Users/mlomb/Desktop/tracked_files_github/bdnf_4su/data/Full BDNF 4SU - salmon/Full BDNF 4SU - salmon/pfam_analysis_1hr.txt")
switch_list_1hr_part2 <- analyzeIUPred2A(switchAnalyzeRlist = switch_list_1hr_part2,
                                      pathToIUPred2AresultFile = "C:/Users/mlomb/Desktop/tracked_files_github/bdnf_4su/data/Full BDNF 4SU - salmon/Full BDNF 4SU - salmon/IUPred_1hr.result")
switch_list_1hr_part2 <- analyzeSignalP(switchAnalyzeRlist = switch_list_1hr_part2,
                                     pathToSignalPresultFile = "C:/Users/mlomb/Desktop/tracked_files_github/bdnf_4su/data/Full BDNF 4SU - salmon/Full BDNF 4SU - salmon/signalIP_1hr.txt")
switch_list_1hr_part2 <- analyzeAlternativeSplicing(switchAnalyzeRlist = switch_list_1hr_part2,
                                                 onlySwitchingGenes = TRUE)
switch_list_1hr_part2 <- analyzeSwitchConsequences(switchAnalyzeRlist = switch_list_1hr_part2,
                                                 consequencesToAnalyze = c('last_exon', "5_utr_length", "3_utr_length","5_utr_seq_similarity", "3_utr_seq_similarity"))
switchPlotTopSwitches(switchAnalyzeRlist = switch_list_1hr_part2, 
            n = 50, filterForConsequences = TRUE, 
            splitFunctionalConsequences = TRUE,
            pathToOutput = "C:/Users/mlomb/Desktop/tracked_files_github/bdnf_4su/splicing plots", 
            fileType = "pdf")
extractConsequenceSummary(switchAnalyzeRlist = switch_list_1hr_part2, 
                plot = TRUE, plotGenes = TRUE, localTheme = ggpubr::theme_pubclean())
extractConsequenceEnrichment(switchAnalyzeRlist = switch_list_1hr_part2, 
                plot = TRUE)
extractSplicingEnrichment(switchAnalyzeRlist = switch_list_1hr_part2, 
                plot = TRUE)
```

5utr 3utr isoform switch genes for categorization

```{r getting genes for shorter last exon 1hr}
switch_list_consequence_1hr <- switch_list_1hr_part2$switchConsequence

switch_list_consequence_1hr %>% 
  filter(switchConsequence == 'Last exon more upstream' | switchConsequence == 'Last exon more downstream') %>% 
  pull(gene_name) %>% 
  unique() %>% 
  clipr::write_clip()
```


2hrs
---------------------------------------------------------
```{r setup 2 hr}
salmon_data_2hr <- importIsoformExpression(parentDir = "C:/Users/mlomb/Desktop/tracked_files_github/bdnf_4su/data/Full BDNF 4SU - salmon/Full BDNF 4SU - salmon/two_hour",addIsofomIdAsColumn = TRUE)

conditions <- gsub('_[1-9][A-Z]_.*', '', colnames(salmon_data_2hr$abundance)[-1])
conditions <- gsub('[1-9][A-Z]_','', conditions)

myDesign_2hr <- data.frame(
  sampleID = colnames(salmon_data_2hr$abundance)[-1],
  condition = conditions)

comparison_to_make = data.frame(condition_1 = "CONTROL", condition_2 = "BDNF")

aSwitchList_2hr <- importRdata(
  isoformCountMatrix = salmon_data_2hr$counts,
  isoformRepExpression = salmon_data_2hr$abundance,
  designMatrix = myDesign_2hr,
  comparisonsToMake = comparison_to_make,
  isoformExonAnnoation = "C:/Users/mlomb/Desktop/tracked_files_github/bdnf_4su/gencode.v40.annotation.gtf.gz",
  isoformNtFasta = "C:/Users/mlomb/Desktop/tracked_files_github/bdnf_4su/gencode.v40.transcripts.fa",
  showProgress = FALSE
)
```


step 1) extract isoform switches and their sequences
```{r step 1 for 2hr}

switch_list_2hr <- isoformSwitchAnalysisPart1(
  switchAnalyzeRlist = aSwitchList_2hr,
  pathToOutput = "C:/Users/mlomb/Desktop/tracked_files_github/bdnf_4su/data/Full BDNF 4SU - salmon/Full BDNF 4SU - salmon",
  outputSequences = TRUE,
  prepareForWebServers = FALSE
)
```

volcano plots 2hrs
```{r volcano 2hrs}
source(here::here('make_volcano_plot.R'))
library(ggplot2)

switch_list_info_2hr_analyzed <- switch_list_2hr_analyzed$isoformSwitchAnalysis

switch_list_info_2hr_analyzed <- switch_list_info_2hr_analyzed %>% 
  mutate(isoform_id = gsub('\\..*',"", isoform_id)) %>% 
  left_join(annotables::grch38_tx2gene, by = c(isoform_id = 'enstxp')) %>% 
  left_join(annotables::grch38 %>%  dplyr::select(symbol, ensgene), by = 'ensgene')

label_significant(switch_list_info_2hr_analyzed, log2FoldCut = 0.2, log10padj = 15)
```

See chunk 12 to do the commented out portion, it doesn't work unless broken up.
```{r step 2: plot isoform switches, 2hr}

switch_list_2hr_filtered <- preFilter(switch_list_2hr,
                                      isoformExpressionCutoff = 1.85) # preFilter to remove lowly expressed features and to get under 500 for the pfam

switch_list_2hr_analyzed <- isoformSwitchTestDEXSeq(
    switchAnalyzeRlist = switch_list_2hr_filtered,
    reduceToSwitchingGenes=TRUE)

switch_list_2hr_fasta <- extractSequence(switch_list_2hr_analyzed,
                                        pathToOutput = "C:/Users/mlomb/Desktop/tracked_files_github/bdnf_4su/data/Full BDNF 4SU - salmon/Full BDNF 4SU - salmon",
                                        writeToFile = TRUE)


##isoform_switch_2hr_step2 <- isoformSwitchAnalysisPart2(switchAnalyzeRlist = switch_list_2hr_filtered n = 10, removeNoncodinORFs = TRUE, pathToCPC2resultFile = "C:/Users/mlomb/Desktop/tracked_files_github/bdnf_4su/data/Full BDNF 4SU - salmon/Full BDNF 4SU - salmon/cpc2_2hr.txt", pathToPFAMresultFile = "C:/Users/mlomb/Desktop/tracked_files_github/bdnf_4su/data/Full BDNF 4SU - salmon/Full BDNF 4SU - salmon/pfam_analysis.txt", pathToSignalPresultFile = "C:/Users/mlomb/Desktop/tracked_files_github/bdnf_4su/data/Full BDNF 4SU - salmon/Full BDNF 4SU - salmon/signal_ip_2hr.txt", pathToIUPred2AresultFile = "C:/Users/mlomb/Desktop/tracked_files_github/bdnf_4su/data/Full BDNF 4SU - salmon/Full BDNF 4SU - salmon/IUPred_1hr.result",outputPlots = TRUE,pathToOutput = "C:/Users/mlomb/Desktop/tracked_files_github/bdnf_4su/data/Full BDNF 4SU - salmon/Full BDNF 4SU - salmon/two_hour", fileType = 'pdf')
```

```{r other plotting for hour 2}

#for GO analysis
switch_list_info_2hr_analyzed %>% 
  filter(padj < 0.1) %>% 
  unique() %>% 
  pull(symbol) %>% 
  clipr::write_clip()

```

go analysis 2 hr significantly changed isoform genes
```{r go 2hr sig isoform switch}
cc_go_2hr_isoform_switch_fp <- file.path("data","Full BDNF 4SU - salmon","Full BDNF 4SU - salmon","cc_go_2hr_isoform_switch.tsv")

cc_go_2hr_isoform_switch <- janitor::clean_names(read_tsv(cc_go_2hr_isoform_switch_fp))

cc_go_2hr_isoform_switch %>%
  slice_max(strength, n = 25) %>%
  mutate(term_description = fct_reorder(term_description,strength)) %>%  
  ggplot(aes(y = strength, fill = false_discovery_rate, x = term_description,size = observed_gene_count)) + 
  geom_point(pch = 21) + 
  coord_flip()+ 
  ggpubr::theme_pubr()+
  ggtitle("CC GO of genes overlapping with significant isoform switching at 2hr")
```

taking the analysis part of isoformswitchanalysispart2() because it doesn't output my graphs properly
```{r recreate isoformswitchanalysispart2()}
switch_list_2hr_part2 <- analyzeCPC2(switchAnalyzeRlist = switch_list_2hr,
                                  pathToCPC2resultFile = "C:/Users/mlomb/Desktop/tracked_files_github/bdnf_4su/data/Full BDNF 4SU - salmon/Full BDNF 4SU - salmon/cpc2_2hr.txt",
                                  codingCutoff = 0.5,
                                  removeNoncodinORFs = TRUE)
switch_list_2hr_part2 <- analyzePFAM(switchAnalyzeRlist = switch_list_2hr_part2,
                                  pathToPFAMresultFile = "C:/Users/mlomb/Desktop/tracked_files_github/bdnf_4su/data/Full BDNF 4SU - salmon/Full BDNF 4SU - salmon/pfam_2hr.txt")
switch_list_2hr_part2 <- analyzeIUPred2A(switchAnalyzeRlist = switch_list_2hr_part2,
                                      pathToIUPred2AresultFile = "C:/Users/mlomb/Desktop/tracked_files_github/bdnf_4su/data/Full BDNF 4SU - salmon/Full BDNF 4SU - salmon/iupred_2hr.result")
switch_list_2hr_part2 <- analyzeSignalP(switchAnalyzeRlist = switch_list_2hr_part2,
                                     pathToSignalPresultFile = "C:/Users/mlomb/Desktop/tracked_files_github/bdnf_4su/data/Full BDNF 4SU - salmon/Full BDNF 4SU - salmon/signal_ip_2hr.txt")
switch_list_2hr_part2 <- analyzeAlternativeSplicing(switchAnalyzeRlist = switch_list_2hr_part2,
                                                 onlySwitchingGenes = TRUE)
switch_list_2hr_part2 <- analyzeSwitchConsequences(switchAnalyzeRlist = switch_list_2hr_part2,
                                                 consequencesToAnalyze = c('last_exon', "5_utr_length", "3_utr_length","5_utr_seq_similarity", "3_utr_seq_similarity"))

switchPlotTopSwitches(switchAnalyzeRlist = switch_list_2hr_part2, 
            n = 50, filterForConsequences = TRUE, 
            splitFunctionalConsequences = TRUE,
            pathToOutput = "C:/Users/mlomb/Desktop/tracked_files_github/bdnf_4su/splicing plots", 
            fileType = "pdf")
extractConsequenceSummary(switchAnalyzeRlist = switch_list_2hr_part2, 
                plot = TRUE, returnResult = TRUE)
extractConsequenceEnrichment(switchAnalyzeRlist = switch_list_2hr_part2, 
                plot = TRUE, returnResult = TRUE)
extractSplicingEnrichment(switchAnalyzeRlist = switch_list_2hr_part2, 
                plot = TRUE, returnResult = TRUE)
```

```{r looping for the switchplots for shorter 3'utr genes}
pdf('shorter_3utr_genes_isoform_switch_2hr.pdf') 
for(g in upstream_last_exon_genes_2hr){
  switchPlot(switch_list_2hr_part2, gene = g)
}
  dev.off()
```


```{r getting a list of genes that go coding > noncoding, 2hr}
switch_list_consequence_2hr <- switch_list_2hr_part2$switchConsequence

switch_list_consequence_2hr %>% 
  filter(switchConsequence == 'Transcript is Noncoding') %>% 
  pull(gene_name) %>% 
  unique() %>% 
  clipr::write_clip()
```

GRAPH OF coding > noncoding genes new/total rna categorizaiton
```{r new categorization of coding >noncoding genes}
coding_to_non_fp<-file.path('coding_to_noncoding_genes_2hr.csv')
coding_to_noncoding_genes <- read_csv(coding_to_non_fp)

palette<- c("#A6CEE3", "#1F78B4", "#33A02C")

full_de_min2 %>% 
  filter(gene_name %in% coding_to_noncoding_genes$x) %>% 
  group_by(new_rna_sig, total_rna_sig) %>% 
  summarize(count = n()) %>% 
  ggplot(aes(x = total_rna_sig, y = count, fill = new_rna_sig))+
    scale_fill_manual(values = palette)+
    geom_col() +
    coord_flip()+
    ggtitle("Categorizing New vs total RNA -\n for genes that switch from coding to noncoding at 2 hours") +
    labs(y = "Number of Genes", 
         x = "Total RNA Categorization",
         fill = "New RNA Categorization")+
  theme(axis.text.y = element_text(angle = 45), legend.position = "bottom", plot.title = element_text(hjust = 0.5), legend.text = element_text(size = 8), legend.title = element_text(size =10))+
  ggpubr::theme_pubr()
```

```{r getting genes that intron retention gain 2hr}
switch_list_consequence_2hr <-isoform_switch_2hr$switchConsequence

switch_list_consequence_2hr %>% 
  filter(switchConsequence == 'Intron retention gain') %>% 
  pull(gene_name) %>% 
  unique() %>% 
  clipr::write_clip()
```

graph of INTRON RETENTION GAIN new/total rna cateorization

```{r categorization of intron retention gain genes}
intron_retention_fp <- file.path('data', 'intron_retained_genes.csv')
intron_retained_genes <- read_csv(intron_retention_fp)

full_de_min2 <- readRDS('full_de_min2.rds')

palette<- c("#A6CEE3", "#1F78B4", "#33A02C")

full_de_min2 %>% 
  filter(gene_name %in% intron_retained_genes$x) %>% 
  group_by(new_rna_sig, total_rna_sig) %>% 
  summarize(count = n()) %>% 
  ggplot(aes(x = total_rna_sig, y = count, fill = new_rna_sig))+
    scale_fill_manual(values = palette)+
    geom_col() +
    coord_flip()+
    ggtitle("Categorizing New vs total RNA -\nfor genes that switch to intron retention gain at 2 hours") +
    labs(y = "Number of Genes", 
         x = "Total RNA Categorization",
         fill = "New RNA Categorization")+
  theme(axis.text.y = element_text(angle = 45), legend.position = "bottom", plot.title = element_text(hjust = 0.5), legend.text = element_text(size = 8), legend.title = element_text(size =10))+
  ggpubr::theme_pubr()

full_de_min2 %>% 
  filter(gene_name %in% intron_retained_genes$x) %>% 
  filter(total_rna_sig == 'upregulated') %>% 
  pull(gene_name) %>% 
  unique() %>% 
  clipr::write_clip()

```

```{r more upstream/downstream last exon, 2hr}
switch_list_2hr_part2 <- readRDS('isoform_switch_2hr_data.rds')
switch_list_consequence_2hr <-switch_list_2hr_part2$switchConsequence

switch_list_consequence_2hr %>% 
  filter(switchConsequence == 'Last exon more upstream'| switchConsequence=='Last exon more downstream') %>% 
  pull(gene_name) %>% 
  unique() %>% 
  clipr::write_clip()

shorter_longer_last_exon_genes_2hr <- readxl::read_xlsx('shorter_longer_last__exon_genes_2hr.xlsx')

switch_list_2hr_part2 <- readRDS('isoform_switch_2hr_5utr_3utr.rds')

switch_list_2hr_consequence <- switch_list_2hr_part2$switchConsequence 
  
switch_list_2hr_consequence %>% 
  filter(switchConsequence == 'Last exon more upstream' | switchConsequence == 'Last exon more downstream')

switchPlot(switch_list_2hr_part2, gene = 'MAP3K13')
switchPlot(switch_list_2hr_part2, gene = 'METTL15')
switchPlot(switch_list_2hr_part2, gene = 'PIK3R4')
```

more upstream/downstream last exon categorization, 6hr
*when i first did this i misunderstood what more upstream/downstream meant and though it had to do with length which is why the csv is named incorrectly
```{r more upstream/downstream last exon, 6hr}
shorter_longer_last_exon_6hr <- readxl::read_excel('6hr_shorter_longer_last_exon_genes.xlsx')

palette<- c("#A6CEE3", "#1F78B4","#B2DF8A", "#33A02C")

full_de_min2 %>% 
  filter(gene_name %in% shorter_longer_last_exon_6hr$x) %>% 
  group_by(new_rna_sig, total_rna_sig) %>% 
  summarize(count = n()) %>% 
  ggplot(aes(x = total_rna_sig, y = count, fill = new_rna_sig))+
    scale_fill_manual(values = palette)+
    geom_col() +
    coord_flip()+
    ggtitle("Categorizing New vs total RNA -\n for genes that switch to more upstream or more downstream last exon at 6 hours") +
    labs(y = "Number of Genes", 
         x = "Total RNA Categorization",
         fill = "New RNA Categorization")+
  theme(axis.text.y = element_text(angle = 45), legend.position = "bottom", plot.title = element_text(hjust = 0.5), legend.text = element_text(size = 8), legend.title = element_text(size =10))
```

GRAPH OF orf > non-orf switch new/total rna categorization

```{r new categorization of orf > non orf genes}
orf_loss_fp <- file.path('data','orf_loss_2hr.csv')
orf_loss_genes <- read_csv(orf_loss_fp)

palette<- c("#A6CEE3", "#1F78B4", "#33A02C")

full_de_min2 %>% 
  filter(gene_name %in% orf_loss_genes$x) %>% 
  group_by(new_rna_sig, total_rna_sig) %>% 
  summarize(count = n()) %>% 
  ggplot(aes(x = total_rna_sig, y = count, fill = new_rna_sig))+
    scale_fill_manual(values = palette)+
    geom_col() +
    coord_flip()+
    ggtitle("Categorizing New vs total RNA -\nfor genes that switch to Complete ORF loss at 2 hours") +
    labs(y = "Number of Genes", 
         x = "Total RNA Categorization",
         fill = "New RNA Categorization")+
  theme(axis.text.y = element_text(angle = 45), legend.position = "bottom", plot.title = element_text(hjust = 0.5), legend.text = element_text(size = 8), legend.title = element_text(size =10))+
  ggpubr::theme_pubr()


full_de_min2 %>% 
  filter(gene_name %in% orf_loss_genes$x) %>% 
  filter(total_rna_sig == 'not_significant' & new_rna_sig == 'bdnf_higher_new_rna') %>% 
  slice_min(mean_diff,n = 10) %>% 
  pull(gene_name) %>% 
  clipr::write_clip()
```

example of intron retention 2hr

```{r ex switchplot intron retention 2hr}
switch_list_2hr_part2 <- readRDS('isoform_switch_2hr_data.rds')
switch_list_2hr_consequence <- switch_list_2hr_part2$switchConsequence

switch_list_2hr_consequence %>% 
  filter(switchConsequence == 'Intron retention gain') %>% 
  pull(gene_name) %>% 
  unique()

switchPlot(switch_list_2hr_part2, gene = 'CASTOR1')
```

example of coding to noncoding 2hr

```{r ex coding to noncoding switchplot 2hr}
switch_list_2hr_part2 <- readRDS('isoform_switch_2hr_data.rds')
switch_list_2hr_consequence <- switch_list_2hr_part2$switchConsequence

switch_list_2hr_consequence %>% 
  filter(switchConsequence == 'Transcript is Noncoding') %>% 
  pull(gene_name) %>% 
  unique()

switchPlot(switch_list_2hr_part2, gene = 'PRKN')
```

loop for shorter 3'utr

```{r loop for shorter 3'utr}
longer_3utr_genes <- readxl::read_xlsx('longer_3utr_genes.xlsx')

pdf('longer_3utr_isoform_switch_2hr.pdf') 
for(g in longer_3utr_genes$x){
  switchPlot(switch_list_2hr_part2, gene = g)
}
  dev.off()
```

2hrs running all possible switch consequences
```{r 2hr all possible switch consequences}
a_switch_list_2hr <- readRDS('aSwitchList_2hr.rds')

switch_list_2hr <- isoformSwitchAnalysisPart1(
  switchAnalyzeRlist = a_switch_list_2hr,
  pathToOutput = "C:/Users/mlomb/Desktop/tracked_files_github/bdnf_4su/data/Full BDNF 4SU - salmon/Full BDNF 4SU - salmon",
  outputSequences = TRUE,
  prepareForWebServers = FALSE
)

switch_list_2hr_filtered <- preFilter(switch_list_2hr,
                                      isoformExpressionCutoff = 1.85) # preFilter to remove lowly expressed features and to get under 500 for the pfam

switch_list_2hr_analyzed <- isoformSwitchTestDEXSeq(
    switchAnalyzeRlist = switch_list_2hr_filtered,
    reduceToSwitchingGenes=TRUE)

switch_list_2hr_fasta <- extractSequence(switch_list_2hr_analyzed,
                                        pathToOutput = "C:/Users/mlomb/Desktop/tracked_files_github/bdnf_4su/data/Full BDNF 4SU - salmon/Full BDNF 4SU - salmon",
                                        writeToFile = TRUE)

switch_list_2hr_part2 <- analyzeCPC2(switchAnalyzeRlist = switch_list_2hr,
                                  pathToCPC2resultFile = "C:/Users/mlomb/Desktop/tracked_files_github/bdnf_4su/data/Full BDNF 4SU - salmon/Full BDNF 4SU - salmon/cpc2_2hr.txt",
                                  codingCutoff = 0.5,
                                  removeNoncodinORFs = TRUE)
switch_list_2hr_part2 <- analyzePFAM(switchAnalyzeRlist = switch_list_2hr_part2,
                                  pathToPFAMresultFile = "C:/Users/mlomb/Desktop/tracked_files_github/bdnf_4su/data/Full BDNF 4SU - salmon/Full BDNF 4SU - salmon/pfam_2hr.txt")
switch_list_2hr_part2 <- analyzeIUPred2A(switchAnalyzeRlist = switch_list_2hr_part2,
                                      pathToIUPred2AresultFile = "C:/Users/mlomb/Desktop/tracked_files_github/bdnf_4su/data/Full BDNF 4SU - salmon/Full BDNF 4SU - salmon/iupred_2hr.result")
switch_list_2hr_part2 <- analyzeSignalP(switchAnalyzeRlist = switch_list_2hr_part2,
                                     pathToSignalPresultFile = "C:/Users/mlomb/Desktop/tracked_files_github/bdnf_4su/data/Full BDNF 4SU - salmon/Full BDNF 4SU - salmon/signal_ip_2hr.txt")
switch_list_2hr_part2 <- analyzeAlternativeSplicing(switchAnalyzeRlist = switch_list_2hr_part2,
                                                 onlySwitchingGenes = TRUE)
switch_list_2hr_part2 <- analyzeSwitchConsequences(switchAnalyzeRlist = switch_list_2hr_part2,
                                                 consequencesToAnalyze = c("tss","tts","last_exon","isoform_length","intron_structure","intron_retention","isoform_class_code","coding_potential","ORF_genomic","ORF_length","5_utr_length","3_utr_length","isoform_seq_similarity","ORF_seq_similarity","5_utr_seq_similarity","3_utr_seq_similarity","NMD_status","domains_identified","genomic_domain_position","domain_length","signal_peptide_identified","IDR_identified","IDR_length","IDR_type"))

switchPlotTopSwitches(switchAnalyzeRlist = switch_list_2hr_part2, 
            n = 50, filterForConsequences = TRUE, 
            splitFunctionalConsequences = TRUE,
            pathToOutput = "C:/Users/mlomb/Desktop/tracked_files_github/bdnf_4su/splicing plots", 
            fileType = "pdf")
extractConsequenceSummary(switchAnalyzeRlist = isoform_switch_2hr,
                          consequencesToAnalyze = c("tss","tts","last_exon","isoform_length","intron_retention","coding_potential","ORF_genomic","5_utr_length","3_utr_length","NMD_status","domains_identified","genomic_domain_position","IDR_identified", 'ORF_length'),
                          removeEmptyConsequence= TRUE,
                plot = TRUE, returnResult = FALSE)
extractConsequenceEnrichment(switchAnalyzeRlist = isoform_switch_2hr, consequencesToAnalyze = c("tss","tts","last_exon","isoform_length","intron_retention","coding_potential","ORF_genomic","5_utr_length","3_utr_length","NMD_status","domains_identified","genomic_domain_position","domain_length","signal_peptide_identified","IDR_identified","IDR_length","IDR_type",'ORF_length'),
                plot = TRUE, returnResult = FALSE)
extractSplicingEnrichment(switchAnalyzeRlist = switch_list_2hr_part2, 
                plot = TRUE, returnResult = TRUE)
```

6hr
------------------------------------------------

setup: getting the data you need to do the rest
```{r setup, 6hr}
salmon_data_6hr <- importIsoformExpression(parentDir = "C:/Users/mlomb/Desktop/tracked_files_github/bdnf_4su/data/Full BDNF 4SU - salmon/Full BDNF 4SU - salmon/six_hour",addIsofomIdAsColumn = TRUE)

conditions <- gsub('_[1-9][A-Z]_.*', '', colnames(salmon_data_6hr$abundance)[-1])
conditions <- gsub('[1-9][A-Z]_','', conditions)

myDesign_6hr <- data.frame(
  sampleID = colnames(salmon_data_6hr$abundance)[-1],
  condition = conditions)

comparison_to_make = data.frame(condition_1 = "CONTROL", condition_2 = "BDNF")

aSwitchList_6hr <- importRdata(
  isoformCountMatrix = salmon_data_6hr$counts,
  isoformRepExpression = salmon_data_6hr$abundance,
  designMatrix = myDesign_6hr,
  comparisonsToMake = comparison_to_make,
  isoformExonAnnoation = "C:/Users/mlomb/Desktop/tracked_files_github/bdnf_4su/gencode.v40.annotation.gtf.gz",
  isoformNtFasta = "C:/Users/mlomb/Desktop/tracked_files_github/bdnf_4su/gencode.v40.transcripts.fa",
  showProgress = FALSE
)
```


step 1) extract isoform switches and their sequences
```{r step 1: extract isoform switches 6hr}

switch_list_6hr <- isoformSwitchAnalysisPart1(
  switchAnalyzeRlist = aSwitchList_6hr,
  pathToOutput = "C:/Users/mlomb/Desktop/tracked_files_github/bdnf_4su/data/Full BDNF 4SU - salmon/Full BDNF 4SU - salmon",
  outputSequences = TRUE,
  prepareForWebServers = FALSE
)

```

pulling the top 10 isoform changes and making some volcano plots
```{r top isoform switches and volcano plots 6hr}

source(here::here('make_volcano_plot.R'))
library(ggplot2)

switch_list_6hr_filtered <- preFilter(switch_list_6hr,
                                      isoformExpressionCutoff = 1.92) # preFilter to remove lowly expressed features

switch_list_6hr_analyzed <- isoformSwitchTestDEXSeq(
    switchAnalyzeRlist = switch_list_6hr_filtered,
    reduceToSwitchingGenes=TRUE)

switch_list_info_6hr_analyze <- switch_list_6hr_analyzed$isoformSwitchAnalysis

switch_list_info_6hr_analyze <- switch_list_info_6hr_analyze %>% 
  mutate(isoform_id = gsub('\\..*',"", isoform_id)) %>% 
  left_join(annotables::grch38_tx2gene, by = c(isoform_id = 'enstxp')) %>% 
  left_join(annotables::grch38 %>%  dplyr::select(symbol, ensgene), by = 'ensgene')

label_significant(switch_list_info_6hr_analyze, log2FoldCut = 0.2)
```


step 2) plot all isoform switches and their annotation
```{r step 2: plot isoform switches, 6hr}

switch_list_6hr_filtered <- preFilter(switch_list_6hr,
                                      isoformExpressionCutoff = 1.2) # preFilter to remove lowly expressed features

switch_list_6hr_analyzed <- isoformSwitchTestDEXSeq(
    switchAnalyzeRlist = switch_list_6hr_filtered,
    reduceToSwitchingGenes=TRUE)

switch_list_6hr_pfam <- extractSequence(switch_list_6hr_analyzed,
                                        pathToOutput = "C:/Users/mlomb/Desktop/tracked_files_github/bdnf_4su/data/Full BDNF 4SU - salmon/Full BDNF 4SU - salmon",
                                        writeToFile = TRUE)


#isoform_switch_1hr_step2 <- isoformSwitchAnalysisPart2(switchAnalyzeRlist = switch_list_1hr_filtered,
#                                                       n = 10,
#                                                       removeNoncodinORFs = TRUE,
#                                                       pathToCPC2resultFile = "C:/Users/mlomb/Desktop/tracked_files_github/bdnf_4su/data/Full BDNF 4SU - salmon/Full BDNF 4SU - salmon/cpc2_isoform_switch.txt",
#                                                      pathToPFAMresultFile = "C:/Users/mlomb/Desktop/tracked_files_github/bdnf_4su/data/Full BDNF 4SU - salmon/Full #BDNF 4SU - salmon/pfam_analysis.txt",
#                                                       pathToSignalPresultFile = "C:/Users/mlomb/Desktop/tracked_files_github/bdnf_4su/data/Full BDNF 4SU - salmon/Full #BDNF 4SU - salmon/signalIP_1hr.txt",
#                                                       pathToIUPred2AresultFile = "C:/Users/mlomb/Desktop/tracked_files_github/bdnf_4su/data/Full BDNF 4SU - #salmon/Full BDNF 4SU - salmon/IUPred_1hr.result",
#                                                       outputPlots = TRUE,
#                                                       pathToOutput = "C:/Users/mlomb/Desktop/tracked_files_github/bdnf_4su/data/Full BDNF 4SU - salmon/Full BDNF 4SU - salmon/one_hour",
#                                                       fileType = 'pdf')
```

```{r pulling apart part 2 analysis, 6hr}
switch_list_6hr_part2 <- analyzeCPC2(switchAnalyzeRlist = switch_list_6hr,
                                  pathToCPC2resultFile = "C:/Users/mlomb/Desktop/tracked_files_github/bdnf_4su/data/Full BDNF 4SU - salmon/Full BDNF 4SU - salmon/result_cpc2_6hr.txt",
                                  codingCutoff = 0.5,
                                  removeNoncodinORFs = TRUE)
switch_list_6hr_part2 <- analyzePFAM(switchAnalyzeRlist = switch_list_6hr_part2,
                                  pathToPFAMresultFile = "C:/Users/mlomb/Desktop/tracked_files_github/bdnf_4su/data/Full BDNF 4SU - salmon/Full BDNF 4SU - salmon/pfam_6hr.txt")
switch_list_6hr_part2 <- analyzeIUPred2A(switchAnalyzeRlist = switch_list_6hr_part2,
                                      pathToIUPred2AresultFile = "C:/Users/mlomb/Desktop/tracked_files_github/bdnf_4su/data/Full BDNF 4SU - salmon/Full BDNF 4SU - salmon/iupred_6hr.result")
switch_list_6hr_part2 <- analyzeSignalP(switchAnalyzeRlist = switch_list_6hr_part2,
                                     pathToSignalPresultFile = "C:/Users/mlomb/Desktop/tracked_files_github/bdnf_4su/data/Full BDNF 4SU - salmon/Full BDNF 4SU - salmon/signal_ip_6hr.txt")
switch_list_6hr_part2 <- analyzeAlternativeSplicing(switchAnalyzeRlist = switch_list_6hr_part2,
                                                 onlySwitchingGenes = TRUE)
switch_list_6hr_part2 <- analyzeSwitchConsequences(switchAnalyzeRlist = switch_list_6hr_part2,
                                                 consequencesToAnalyze = c('last_exon', "5_utr_length", "3_utr_length","5_utr_seq_similarity", "3_utr_seq_similarity"))

switchPlotTopSwitches(switchAnalyzeRlist = switch_list_6hr_part2, 
            n = 50, filterForConsequences = TRUE, 
            splitFunctionalConsequences = TRUE,
            pathToOutput = "C:/Users/mlomb/Desktop/tracked_files_github/bdnf_4su/splicing plots", 
            fileType = "pdf")
extractConsequenceSummary(switchAnalyzeRlist = switch_list_6hr_part2, 
                plot = TRUE, plotGenes = TRUE, localTheme = ggpubr::theme_pubclean())
extractConsequenceEnrichment(switchAnalyzeRlist = switch_list_6hr_part2, 
                plot = TRUE)
extractSplicingEnrichment(switchAnalyzeRlist = switch_list_6hr_part2, 
                plot = TRUE)
```

```{r looping the last exon changes for 6hr into a pdf}
pdf('shorter_longer_last_exon_switch_plots_6hr.pdf') 
for(g in shorter_longer_last_exon_genes_6hr$x){
  switchPlot(switch_list_6hr_part2, gene = g)
}
  dev.off()
```


```{r looking at the isoform switch for the upregulated coding > noncoding genes at 2hr}


isoform_switch_2hr <- readRDS('isoform_switch_2hr_data.rds')

#from upregulated total rna for categorization chart

switchPlot(isoform_switch_2hr, 
           gene = 'EPHB6')

switchPlot(isoform_switch_2hr, gene = 'SV2B')

switchPlot(isoform_switch_2hr, gene = 'DNMT1')

switchPlot(isoform_switch_2hr, gene = 'PTPRE')

switchPlot(isoform_switch_2hr, gene = 'MGAT4C')

switchPlot(isoform_switch_2hr, gene = 'LPP')

switchPlot(isoform_switch_2hr, gene = 'TSPAN9')

switchPlot(isoform_switch_2hr, gene = 'SLC25A17')

switchPlot(isoform_switch_2hr, gene = 'RAB31')

switchPlot(isoform_switch_2hr, gene = 'TSPAN18')

```

```{r examples of isoform switch resulting in orf_loss}
switch_consequence_2hr <- isoform_switch_2hr$switchConsequence

orf_loss <- switch_consequence_2hr %>% 
  filter(switchConsequence == "Complete ORF loss") %>% 
  pull(gene_name) %>% 
  unique() %>% 
  clipr::write_clip()

#from upregulated total rna from the categorization chart

switchPlot(isoform_switch_2hr, gene = 'RAP1GAP')

switchPlot(isoform_switch_2hr, gene = 'LPP')

switchPlot(isoform_switch_2hr, gene = 'EPHB6')

switchPlot(isoform_switch_2hr, gene = 'EBF2')

switchPlot(isoform_switch_2hr, gene = 'PTPRE')

switchPlot(isoform_switch_2hr, gene = 'TSPAN9')

switchPlot(isoform_switch_2hr, gene = 'MGAT4C')

switchPlot(isoform_switch_2hr, gene = 'SV2B')

switchPlot(isoform_switch_2hr, gene = 'SLC25A17')

switchPlot(isoform_switch_2hr, gene = 'TSPAN18')

switchPlot(isoform_switch_2hr, gene = 'RAB31')

#from not significant total rna with high fraction new from cat chart

switchPlot(isoform_switch_2hr, gene = 'PIK3R4')

switchPlot(isoform_switch_2hr, gene = 'ZNF136')

switchPlot(isoform_switch_2hr, gene = 'E2F4') #**

switchPlot(isoform_switch_2hr, gene = 'RASSF5')

switchPlot(isoform_switch_2hr, gene = 'TTF1')

switchPlot(isoform_switch_2hr, gene = 'ZNF121')

switchPlot(isoform_switch_2hr, gene = 'URB2')

switchPlot(isoform_switch_2hr, gene = 'MTERF4')

switchPlot(isoform_switch_2hr, gene = 'KIF26B')

switchPlot(isoform_switch_2hr, gene = 'MED12L')
```

^top 10 genes (based on mean_diff) from the not significant total with higher fraction new that result in ORF loss


#getting a 3way venn of the genes with ORF loss, coding>noncoding switch, and last exon more upstream for 2hr bdnf vs control
```{r venn diagram of overlap between isoform switches}

my_isoform_switch <- readRDS('isoform_switch_all_consequences_2hr.rds')

switch_consequence <- my_isoform_switch$switchConsequence

orf_loss <- switch_consequence %>% 
  filter(switchConsequence == 'Complete ORF loss') %>% 
  pull(gene_name) %>% 
  unique()

noncoding <- switch_consequence %>% 
  filter(switchConsequence == 'Transcript is Noncoding') %>% 
  pull(gene_name) %>% 
  unique()

upstream_last_exon <- switch_consequence %>% 
  filter(switchConsequence == 'Last exon more upstream') %>% 
  pull(gene_name) %>% 
  unique()

library(eulerr)

my_list <- list(orf_loss = orf_loss,
                noncoding = noncoding,
                upstream_last_exon = upstream_last_exon)

my_venn <- plot(euler(my_list), quantities = TRUE)

#getting the 94 genes that just have more upstream_last_exon and no orf_loss or noncoding

only_upstream_last_exon <- switch_consequence %>% 
  filter(switchConsequence == 'Last exon more upstream') %>% 
  filter(!(gene_name %in% orf_loss)) %>% 
  filter(!(gene_name %in% noncoding))%>% 
  pull(gene_name) %>% 
  unique() %>% 
  clipr::write_clip()
```

```{r just longer last exon, 6hr}
longer_last_exon_6hr <- readxl::read_excel('longer_last_exon_genes_6hr.xlsx')

palette<- c("#A6CEE3", "#1F78B4", "#33A02C")

full_de_min2 %>% 
  filter(gene_name %in% longer_last_exon_6hr$x) %>% 
  group_by(new_rna_sig, total_rna_sig) %>% 
  summarize(count = n()) %>% 
  ggplot(aes(x = total_rna_sig, y = count, fill = new_rna_sig))+
    scale_fill_manual(values = palette)+
    geom_col() +
    coord_flip()+
    ggtitle("Categorizing New vs total RNA -\n for genes that switch to longer last exon at 6 hours") +
    labs(y = "Number of Genes", 
         x = "Total RNA Categorization",
         fill = "New RNA Categorization")+
  theme(axis.text.y = element_text(angle = 45), legend.position = "bottom", plot.title = element_text(hjust = 0.5), legend.text = element_text(size = 8), legend.title = element_text(size =10))
```


upstream last exon genes for 1 and 2hrs now 

```{r upstream last exon 2hr}
upstream_last_exon <- readxl::read_xlsx('upstream_last_exon_genes_2hr.xlsx')

palette<- c("#A6CEE3", "#1F78B4","#B2DF8A", "#33A02C")

full_de_min2 %>% 
  filter(gene_name %in% upstream_last_exon$x) %>% 
  group_by(new_rna_sig, total_rna_sig) %>% 
  summarize(count = n()) %>% 
  ggplot(aes(x = total_rna_sig, y = count, fill = new_rna_sig))+
    scale_fill_manual(values = palette)+
    geom_col() +
    coord_flip()+
    ggtitle("Categorizing New vs total RNA -\nfor genes that switch to more upstream last exon at 2 hours") +
    labs(y = "Number of Genes", 
         x = "Total RNA Categorization",
         fill = "New RNA Categorization")+
  theme(axis.text.y = element_text(angle = 45), legend.position = "bottom", plot.title = element_text(hjust = 0.5), legend.text = element_text(size = 8), legend.title = element_text(size =10))+
  ggpubr::theme_pubr()
```